services/horizon/internal/actions/helpers.go

```
@@ -701,3 +760,62 @@ func FullURL(ctx context.Context) *url.URL {
 	}
 	return url
 }
+
+// Note from chi: it is a good idea to set a Decoder instance as a package
+// global, because it caches meta-data about structs, and an instance can be
+// shared safely:
+var decoder = schema.NewDecoder()
+
+// GetParams fills a struct with values read from a request's query parameters.
+func GetParams(dst interface{}, r *http.Request) error {
+	query := r.URL.Query()
+
+	// Merge chi's URLParams with URL Query Params. Given
+	// `/accounts/{account_id}/transactions?foo=bar`, chi's URLParams will
+	// contain `account_id` and URL Query params will contain `foo`.
+	if rctx := chi.RouteContext(r.Context()); rctx != nil {
+		for _, key := range rctx.URLParams.Keys {
+			query.Set(key, rctx.URLParam(key))
+		}
+	}
+
+	if err := decoder.Decode(dst, query); err != nil {
+		return err
+	}
+
+	if _, err := govalidator.ValidateStruct(dst); err != nil {
+		fields := govalidator.ErrorsByField(err)
+
+		for field, message := range fields {
```

the field value, would be as defined in the struct, but we want to actually report the error as defined in the query params.  so instead of `BuyingAssetCode` it will use the value defined in the schema which was the one passed in the request `buying_asset_code`